test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -48,22 +48,22 @@
        :jruby-puppet {:max-active-instances 4}}
       (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
             context (tk-services/service-context jruby-service)
-            pool-context (:pool-context context)
-            pool (jruby-core/get-pool pool-context)]
-        (jruby-testutils/reduce-over-jrubies! pool 4 #(format "InstanceID = %s" %))
+            pool-context (:pool-context context)]
+        (jruby-testutils/reduce-over-jrubies! pool-context 4 #(format "InstanceID = %s" %))
         (is (= #{0 1 2 3}
-               (-> (jruby-testutils/reduce-over-jrubies! pool 4 (constantly "InstanceID"))
+               (-> (jruby-testutils/reduce-over-jrubies! pool-context 4 (constantly "InstanceID"))
                    set)))
         (let [response (http-client/delete
                          "https://localhost:8140/puppet-admin-api/v1/jruby-pool"
                          ssl-request-options)]
           (is (= 204 (:status response))))
         ; wait until the flush is complete
-        (await (:pool-agent context))
-        (let [new-pool (jruby-core/get-pool pool-context)]
-          (is (every? true?
-                      (jruby-testutils/reduce-over-jrubies!
-                        new-pool
-                        4
-                        (constantly
-                          "begin; InstanceID; false; rescue NameError; true; end")))))))))
+        (await (:pool-agent pool-context))
+        (is (every? true?
+                    (jruby-testutils/reduce-over-jrubies!
+                      pool-context
+                      4
+                      (constantly
+                        "begin; InstanceID; false; rescue NameError; true; end"))))))))
+
+;; TODO: test flush instance while pool flush is in progress
```

Comment:
I agree this would be a good one to have, and that it feels like an "integration" kind of test.